packages/edit-site/src/components/style-book/index.js

```diff
@@ -156,65 +245,133 @@ function StyleBook( { isSelected, onSelect, onClose } ) {
+									__(
+										'Examples of blocks in the %s category'
+									),
+									tab.title
+								) }
+								isSelected={ isSelected }
+								onSelect={ onSelect }
+							/>
+						</Iframe>
 					) }
 				</TabPanel>
 			</section>
 		</StyleBookFill>
 	);
 }
 
-const Examples = memo( ( { examples, category, isSelected, onSelect } ) => (
-	<div className="edit-site-style-book__examples">
-		{ examples
-			.filter( ( example ) => example.category === category )
-			.map( ( example ) => (
-				<Example
-					key={ example.name }
-					title={ example.title }
-					blocks={ example.blocks }
-					isSelected={ isSelected( example.name ) }
-					onClick={ () => {
-						onSelect( example.name );
-					} }
-				/>
-			) ) }
-	</div>
-) );
-
-const Example = memo( ( { title, blocks, isSelected, onClick } ) => (
-	<button
-		className={ classnames( 'edit-site-style-book__example', {
-			'is-selected': isSelected,
-		} ) }
-		aria-label={ sprintf(
-			// translators: %s: Title of a block, e.g. Heading.
-			__( 'Open %s styles in Styles panel' ),
-			title
-		) }
-		onClick={ onClick }
-	>
-		<span className="edit-site-style-book__example-title">{ title }</span>
-		<div className="edit-site-style-book__example-preview">
-			<BlockPreview
-				blocks={ blocks }
-				viewportWidth={ 0 }
-				additionalStyles={ [
-					{
-						css:
-							'.wp-block:first-child { margin-top: 0; }' +
-							'.wp-block:last-child { margin-bottom: 0; }',
-					},
-				] }
-			/>
-		</div>
-	</button>
-) );
+const Examples = memo(
+	( { className, examples, category, label, isSelected, onSelect } ) => {
+		const composite = useCompositeState( { orientation: 'vertical' } );
+		return (
+			<Composite
+				{ ...composite }
+				className={ className }
+				aria-label={ label }
+			>
+				{ examples
+					.filter( ( example ) => example.category === category )
+					.map( ( example, index ) => (
+						<Example
+							key={ example.name }
+							id={ `example-${ index }` }
+							composite={ composite }
+							title={ example.title }
+							blocks={ example.blocks }
+							isSelected={ isSelected( example.name ) }
+							onClick={ () => {
+								onSelect( example.name );
+							} }
+						/>
+					) ) }
+			</Composite>
+		);
+	}
+);
+
+const Example = memo(
+	( { composite, id, title, blocks, isSelected, onClick } ) => {
+		const originalSettings = useSelect(
+			( select ) => select( blockEditorStore ).getSettings(),
+			[]
+		);
+		const settings = useMemo(
+			() => ( { ...originalSettings, __unstableIsPreviewMode: true } ),
+			[ originalSettings ]
+		);
+
+		// Cache the list of blocks to avoid additional processing when the component is re-rendered.
+		const renderedBlocks = useMemo(
+			() => ( Array.isArray( blocks ) ? blocks : [ blocks ] ),
+			[ blocks ]
+		);
+
+		return (
+			<CompositeItem
+				{ ...composite }
+				className={ classnames( 'edit-site-style-book__example', {
+					'is-selected': isSelected,
+				} ) }
+				id={ id }
```

Comment:
Is the id being used for anything?

Reply:
Yep! It's a little weird, but I followed the suggestions from [the reakit docs here](https://reakit.io/docs/composite/#performance) that recommend it as a performance enhancement when using `CompositeItem` components when you're concerned about performance. I figured since we're dealing with previews and we want to avoid re-renders where we can, it'd be a good one to include. It's the first time I've used it, though, so very happy for feedback if I've missed anything 😄